### PR TITLE
fix: hack to handle MM account changes

### PR DESCRIPTION
**Pull request by reidrankin**
## Description

Hack to handle MetaMask account changes

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

### CODE REVIEW

**Review by elmutt — APPROVED** [review]

**Review by cjthompson — COMMENTED** [review]
I don't think this is even the right place to put this.

**Comment by cjthompson** [comment]
Lint error

File: src/context/WalletProvider/MetaMask/components/Connect.tsx [hunk 1]
```diff
@@ -27,6 +27,11 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
     setError(null)
     setLoading(true)
     if (state.adapters && state.adapters?.has(KeyManager.MetaMask)) {
+      // Hack to handle MetaMask account changes
+      //TODO: handle this properly
+      window.ethereum?.on?.('accountsChanged', () => window.location.reload())
+      window.ethereum?.on?.('chainChanged', () => window.location.reload())
```
Comment (cjthompson) [inline src/context/WalletProvider/MetaMask/components/Connect.tsx:33]: This is terrible.

File: src/context/WalletProvider/MetaMask/components/Connect.tsx [hunk 2]
```diff
@@ -54,13 +60,20 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
     setLoading(false)
   }
 
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setIsMainnet(window.ethereum?.chainId === '0x1')
+    }, 100)
+    return () => clearInterval(interval)
+  }, [setIsMainnet])
```
Comment (cjthompson) [inline src/context/WalletProvider/MetaMask/components/Connect.tsx:68]: I think it's much better to put it on line 39/45.

**Review by cjthompson — APPROVED** [review]

**Pull request merged**